lib/language_pack/rails5.rb

```diff
@@ -30,4 +30,55 @@ def default_config_vars
   def install_plugins
     # do not install plugins, do not call super, do not warn
   end
+
+  def best_practice_warnings
+    return unless bundler.has_gem?("activestorage")
+    return unless File.exist("config/storage.yml")
+
+    warn_local_storage if local_storage?
+    warn_no_ffmpeg     if needs_ffmpeg?
+  end
+
+  private
+    def has_ffmpeg?
+      path = run("which ffmpeg")
+      return path && $?.success?
+    end
+
+    def needs_ffmpeg?
+      !has_ffmpeg?
+    end
+
+    def local_storage?
+      out = run!('bin/rails runner "puts %Q{heroku_detecting_active_storage_config=#{Rails.application.config.active_storage.service}}"')
+      out =~ /heroku_detecting_active_storage_config=local/
+    end
+
+    def warn_local_storage
+      mcount("warn.activestorage.local_storage")
+      warn(<<-WARNING)
+You set your `config.active_storage.service` to :local in production.
+If you are uploading files to this app, they will not persist after the app
+is restarted. Heroku applications have an ephemeral files system. To
```

Comment:
after the app is restarted, on other web dynos, or in one off dynos.

People don't get this, and I guarantee we'll see tickets about it.

Reply:
Updated. Also I added more words in the linked article.